src/backend/optimizer/plan/createplan.c

```diff
@@ -514,8 +514,7 @@ create_scan_plan(PlannerInfo *root, Path *best_path)
 	/* Decorate the top node of the plan with a Flow node. */
 	plan->flow = cdbpathtoplan_create_flow(root,
 										   best_path->locus,
-								best_path->parent ? best_path->parent->relids
-										   : NULL,
+										   best_path->parent->relids,
```

Comment:
Here `best_path->parent` cannot be NULL, because first `rel = best_path->parent` and a little later `scan_clauses = rel->baserestrictinfo`.